### DISCUSSION TITLE
About trans_nullmodel

### DISCUSSION THREAD

**Discussion by jackqio** [ChiLiubio/microeco#290] [Q&A]
When I used the trans_nullmodel function, for example, and set the code as "test <- trans_nullmodel$new(dataset, group = "Group", select_group = "CY", filter_thres = 0.0005)", the R always said "Error in `[.data.frame`(., , group) :Undefined columns are selected". However, the column name which I gave is right, and I don't where is the code wrong. Could you give me a help?

**Comment by ChiLiubio** [comment]
Hi. Could you please attach your data so that I can reproduce your issue? To save the dataset (your microtable object), please follow the steps in the tutorial (https://chiliubio.github.io/microeco_tutorial/notes.html#save-function) and attach the compressed object.

**Reply by jackqio** [reply]
Hi. When I make a sub data for a test, the problem has disappeared. I delete some important data column in sample_table for confidentiality, and the code has worked somehow. Could the reason for the error be that the sample_table has too many columns? But I still upload a sub data for your test and the code "test <- trans_nullmodel$new(group_S, group = "Group1", select_group = "CY", filter_thres = 0.0005)" still  can't work for this sub data. Thanks for your help.
[test_data.zip](https://github.com/ChiLiubio/microeco/files/13256646/test_data.zip)

**Reply by ChiLiubio** [reply]
Hi. I find it is a bug. I will fix it in the next version. Thanks! It is perfect to extract data before using this class.
